src/engine/keycontrol.cpp

```diff
@@ -212,12 +217,16 @@ void KeyControl::updateRate() {
         // !bKeylock
         if (m_pitchRateInfo.keylock) {
             // Disabling Keylock
-            if (m_keylockMode->get() == kLockCurrentKey) {
-                // reset to linear pitch
-                m_pitchRateInfo.pitchTweakRatio = 1.0;
-                // For not resetting to linear pitch:
-                // Adopt speedPitchRatio change as pitchTweakRatio
-                //pitchRateInfo.pitchTweakRatio *= (m_speedSliderPitchRatio / pitchRateInfo.tempoRatio);
+            // If "Reset (unlocked) Key" is enabled
+            if (m_keyunlockMode->get() == kResetUnlockedKey) {
+                // If 'current' aka 'not original' key was locked
+                if (m_keylockMode->get() == kLockCurrentKey) {
```

Comment:
In this case the Key knob is cranked, right? Original "Reset Key" reset to this cranked key at rate = 1.  Is there a use case for this? 

The user may use it like this: 
* match key 
* enable key-lock 
* match beat 
* Unlock key ... what will happen ?? The key jumps to rather random key now, by adding the offset introduced by the rate slider. 

I think in the "Keep Key" case, the offset should be applied to the key knob by turning it.